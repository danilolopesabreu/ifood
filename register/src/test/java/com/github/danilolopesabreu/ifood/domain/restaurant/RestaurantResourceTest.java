package com.github.danilolopesabreu.ifood.domain.restaurant;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.danilolopesabreu.ifood.infrastructure.DataBaseLifeCycle;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(DataBaseLifeCycle.class)
class RestaurantResourceTest {

	@Test
	@DataSet(value = "restaurant.yml")
	void testListRestaurants() {
		String result = 
				RestAssured
					.given()
						.when().get("/restaurants")
						.then().statusCode(200)
						.extract().asString();
		
		Approvals.verifyJson(result);
	}

}
