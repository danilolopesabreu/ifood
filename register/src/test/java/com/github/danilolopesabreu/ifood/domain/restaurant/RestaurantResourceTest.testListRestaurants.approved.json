[
  {
    "location": {
      "latitude": 10.0,
      "longitude": 20.0
    },
    "name": "test NAME qwerty 66",
    "owner": "man in the box"
  }
]